src/kv/deserialise.h

```diff
@@ -139,6 +139,12 @@ namespace kv
         success = ApplyResult::PASS_SNAPSHOT_EVIDENCE;
       }
 
+      search = changes.find(ccf::Tables::ENCRYPTED_PAST_LEDGER_SECRET);
+      if (search != changes.end())
+      {
+        success = ApplyResult::PASS_ENCRYPTED_PAST_LEDGER_SECRET;
```

Comment:
Do we write to this table in transactions that touch multiple tables? This return scheme originally worked for `PASS_SIGNATURES`, which worked because signatures is a special table with a bunch of other requirements on it. I'm wary of repeating this pattern for too many tables, especially if we're not checking that there's only one table/value written.

Probably out of scope for this PR but: since we now split `deserialise` from application, and its returned value is really the raw `changes`, we could remove all of these enum values and just return `PASS`/`FAIL`, and leave the "did this produce some tables I care about" decision up to the caller.

Reply:
Agreed on the broader store's `deserialise()` refactor (I've raised https://github.com/microsoft/CCF/issues/2208).

Currently, the encrypted past ledger secret map isn't the only map touched by this transaction but the other maps touched aren't considered in the deserialise return code, so I think this change is safe for now. If in the future another map of interest is touched by the rekey transaction, the unit test for historical queries will fail.

Reply:
Why can't this be done with a hook?

Reply:
With the new hook interface, it seems that we can use them but I'd rather do that in a separate PR.

Reply:
Separate PR is fine of course, it's the trend I'm worried about.